File: database/database-api-model/src/main/java/io/ehdev/conrad/database/model/ApiParameterContainer.java
package io.ehdev.conrad.database.model;

import io.ehdev.conrad.database.model.permission.ApiTokenAuthentication;
import io.ehdev.conrad.database.model.project.ApiRepoModel;

public class ApiParameterContainer implements ApiRepoModel {
    private final ApiTokenAuthentication user;
    private final String projectName;
    private final String repoName;

    public ApiParameterContainer(ApiTokenAuthentication user, String projectName, String repoName) {
        this.user = user;
        this.projectName = projectName;
        this.repoName = repoName;
    }

    @Override
    public String getProjectName() {
        return projectName;
    }

    @Override
    public String getRepoName() {
        return repoName;
    }

    @Override
    public String getMergedName() {
        return projectName + "/" + repoName;
    }

    public ApiTokenAuthentication getUser() {
        return user;
    }
}

File: database/database-api-model/src/main/kotlin/io/ehdev/conrad/database/model/ApiParameterContainer.kt
package io.ehdev.conrad.database.model

import io.ehdev.conrad.database.model.project.ApiRepoModel
import io.ehdev.conrad.database.model.user.ApiUser

class ApiParameterContainer(val user: ApiUser?,
                            override val projectName: String?,
                            override val repoName: String?) : ApiRepoModel {

    constructor(user: ApiUser?, repo: ApiRepoModel) : this(user, repo.projectName, repo.projectName)

//    constructor() : this(null, null, null)
}
